CHANGES.md

```diff
@@ -1,6 +1,478 @@
+* improved `--auto-aggregation` to make decisions based on `localAccess` calls
+* improved `--auto-aggregation` to leverage accesses to non-distributed arrays
+* optimized `allLocalesBarrier()`, particularly for InfiniBand systems
+* parallelized assignments of large `bytes` copies in the 'ZMQ' module
+* optimized performance of `string` operations when `--checks` are not enabled
+* optimized the performance of the 'Sort' module's quicksort, used by `sort()`
+* improved fabric selection in `ofi` by defaulting to a large fixed heap
+* added data segment, heap, and stack memory to standard `ofi` registrations
+
+Compilation-Time / Generated Code Improvements
+----------------------------------------------
+* generally speaking, compilation times improved due to the default use of LLVM
+* reduced the amount of code generated for `forall` loops due to fast-followers
+
+Memory Improvements
+-------------------
+* reduced memory fragmentation for configurations that require a fixed heap
+
+Documentation
+-------------
+* added a primer example on C interoperability  
+  (see https://chapel-lang.org/docs/1.25/primers/interopWithC.html)
+* added documentation for how to define operator methods for inheritance  
+  (see https://chapel-lang.org/docs/1.25/technotes/operatorMethods.html#operator-methods-and-classes)
+* added `try...catch` and open-interval ranges to the Quick Reference document
+* documented `locale.runningTasks()`  
+  (see https://chapel-lang.org/docs/1.25/builtins/ChapelLocale.html#ChapelLocale.locale.runningTasks)
+* added new documentation for several symbols in the 'BigInteger' module  
+  (see https://chapel-lang.org/docs/1.25/modules/standard/BigInteger.html)
+* documented the 'CommDiagnostics' fields related to `--cache-remote`  
+  (see https://chapel-lang.org/docs/1.25/modules/standard/CommDiagnostics.html#CommDiagnostics.chpl_commDiagnostics.cache_get_hits)
+* improved the docs regarding which types can be used with `sync`/`single`  
+  (see https://chapel-lang.org/docs/1.25/language/spec/task-parallelism-and-synchronization.html#synchronization-variables)
+* added `throws` documentation to `compile()` in the 'Regex' module  
+  (see https://chapel-lang.org/docs/1.25/modules/standard/Regex.html#Regex.compile)
+* fixed the formatting of double-dash arguments on the online `chpl` man page  
+  (see https://chapel-lang.org/docs/1.25/usingchapel/man.html)
+* described the `make check` target in the 'Building Chapel' documentation  
+  (see https://chapel-lang.org/docs/1.25/usingchapel/building.html#makefile-targets)
+* improved the language specification's formatting of reserved keywords  
+  (see https://chapel-lang.org/docs/1.25/language/spec/lexical-structure.html#keywords)
+* modestly improved the documentation for the 'IO' module
+* fixed the formatting of a list in the 'classes' primer
+* improved the docs for various library routines to reflect return types
+* fixed various typos in the documentation
```

Comment:
* updated the prerequisites documentation page to better show LLVM prerequisites
  (see https://chapel-lang.org/docs/main/usingchapel/prereqs.html#readme-prereqs)
* updated the user-defined reductions spec section to link to example
  (see https://chapel-lang.org/docs/main/language/spec/user-defined-reductions-and-scans.html)